### PR TITLE
Help users understand metric score cards

### LINKED ISSUE #652: Describe the card's possible meter visuals
For those unfamiliar with `{riskmetric}` or metric scores, and given the recent merge of #634, it seems we may need some sort of legend that could help explain these graphics. Per discussion at today's dev meetup, it would seem that perhaps this would be best suited in both the `introjs` steps and also the `Assessment Criteria` tab. There is a lot of explain:

- [ ] No meter, assessment not found warning sign
- [ ] Meter displaying 0
- [ ] Meter displaying 1
- [ ] Meter between 0 and 1
- [ ] Meter displaying NA (assessment value exists, but no score)

Also, perhaps we should mention how the scores work (i.e. that 1 is bad and indicates high risk while 0 is good and indicates low risk).

### PR DESCRIPTION
**Pull request by aclark02-arcus**
Closes #652 
This PR does a number of things to help clarify the metric risk scores. (1) First, it adds tooltips to all the `riskmetric` cards. Then it (2) adds a modal dialog that acts as a legend which provides additional context on how to interpret these assessment & score values. Last, it (30 links back to `riskmetric` documentation on the 'About' Tab. Images below.

## 1. `bslib` tooltips
image

Note: this isn't working as intended for the hyperlinked cards (Dependencies & Reverse Dependencies). I would ultimately like to maintain these hyperlinks, but render the clickable area to just the title and assessment value instead of the whole card.

## 2. Legend
image

Note, the button ✅ will only show in a metricBox metricGrid where a `riskmetric` card exists. 

## 3. References users back to `riskmetric` documentation
image

### CODE REVIEW

**Comment by jthompson-arcus** [comment]
@aclark02-arcus here is a potential solution to the problem you had with tooltips for the cards that redirect (i.e. Dependencies and Reverse Dependencies). The only downside is that the `title` element has been removed, but I think it could be added back in easily enough.
[patch.txt](https://github.com/pharmaR/riskassessment/files/15193083/patch.txt)

**Comment by jthompson-arcus** [comment]
Snipping tool does not capture mouse pointer, but it would look something like this when hovering on the metric label.

image

**Review by narayanan-iyer-pfizer — DISMISSED** [review]

**Comment by aclark02-arcus** [comment]
> What are the opinions on this kind of meter?

@narayanan-iyer-pfizer, it's funny you bring that up, because that's actually where I first started when building these metric card visuals. See [this comment](https://github.com/pharmaR/riskassessment/pull/634#issuecomment-1691904298) which showcases a POC. Overall, we found them to be a bit "busy" aesthetically and hard to tell when things were passing/failing, which is why we opted for a more simple meter.

**Comment by aclark02-arcus** [comment]
Hi @jthompson-arcus, I adopted your patch. I think it works well! Thanks for solving that riddle 👍🏼 
If all else looks good, please push this one into `dev`.

**Comment by aclark02-arcus** [comment]
Oh darn, I just realized that I haven't added these changes into the html reports. Converting to draft for now.

**Comment by aclark02-arcus** [comment]
Okey dokey @jthompson-arcus, I fixed the issue with html reports. This is ready for review (and hopefully merge).

File: R/mod_legend.R [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          div(style = "display: flex; align-items: center;",
+            icon("triangle-exclamation", class = "text-warning", verify_fa = FALSE, style = "font-size:65px; padding-left: 10px;"),
+            div(style = "position: absolute; left: 8rem; padding-right: 20px;", "A 'Not Found' indicates there was no assessment value derived from {riskmetric}'s fetch for this information. This is usually due to the reference sourced used when loading the package into the database. For example, {riskmetric} cannot fetch testing coverage when using the CRAN Remote reference source. Since there is no assessment value, there is also no score. As such, this score will not impact the package's overall risk score.")),
+          br(),br(),br(),
+          div(style = "display: flex; align-items: center;",
+              metric_gauge("NA"),
+              div(style = "position: absolute; left: 8rem; padding-right: 20px;", "An 'NA' indicates an assessment value exists for {riskmetric}, but there is no score recorded. This is usually an intention practice performed by {riskmetric}'s development team when scoring such a metric is ambiguous or strange, thus a score is not applicable")),
```
Comment (jthompson-arcus) [inline R/mod_legend.R:49]:
```suggestion
              div(style = "position: absolute; left: 8rem; padding-right: 20px;", "An 'NA' indicates an assessment value exists for {riskmetric}, but there is no score recorded. This is usually an intentional practice performed by {riskmetric}'s development team when scoring such a metric is ambiguous or strange, thus a score is not applicable")),
```

File: R/mod_legend.R [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          br(),
+          div(style = "display: flex; align-items: center;",
+              metric_gauge("0"),
+              div(style = "position: absolute; left: 8rem; padding-right: 20px;","A score of 1 indicates the highest risk possible based on the {riskmetric} assessment value.")),
+          br(),
+          div(style = "display: flex; align-items: center;",
+              metric_gauge(".50"),
+              div(style = "position: absolute; left: 8rem; padding-right: 20px;", "A Score close to 1 indicates the {riskmetric} assessment value is high risk, where scores closer to 0 inidicate lower risk")),
```
Comment (jthompson-arcus) [inline R/mod_legend.R:57]:
```suggestion
              div(style = "position: absolute; left: 8rem; padding-right: 20px;", "A score close to 1 indicates the {riskmetric} assessment value is high risk, where scores closer to 0 inidicate lower risk")),
```

**Comment by aclark02-arcus** [comment]
Hey @jthompson-arcus, I accepted your changes and also made a few small edits to the modal with my latest commit:
* grammar updates
* added footer to acknowledge how we are taking the inverse of `riskmetric` assessment scores
* for assessment 'not found' visual, I added hyperlink to `riskmetric::pkg_ref` for more info.

This is the result:

image

File: dev/run_dev.R
Comment (jthompson-arcus) [inline dev/run_dev.R:1]: Can these comment additions be removed?

File: tests/testthat/test-metric_gauge.R [hunk 3]
```diff
@@ -31,3 +31,28 @@ test_that("metric_gauge() 'value' working...", {
...
+    # metric_gauge(".25"), br(),
+    metric_gauge(".50"), br(),
+    metric_gauge("1")
+  )
+  server <- function(input, output, session) {}
+  shinyApp(ui, server)
+}
+
+
+
```
Comment (jthompson-arcus) [inline tests/testthat/test-metric_gauge.R:58]: I kind of question the utility of adding this code to the test files. I think if we want to include some little code like this, it should be included in the `R/` folder.

File: dev/run_dev.R
Comment (aclark02-arcus) [reply dev/run_dev.R:1]: Do they have to? lol. I like clean code too, but if they're deleted I'll surely forget the name of the function to load packages outside the app.

File: tests/testthat/test-metric_gauge.R (hunk 3 above)
Comment (Jeff-Thompson12) [reply tests/testthat/test-metric_gauge.R:58]: I wasn't even thinking about `@examples`. I was thinking about an non-exported function. Just feels weird to have it in the testing files.

File: dev/run_dev.R
Comment (Jeff-Thompson12) [reply dev/run_dev.R:1]: Fine

File: tests/testthat/test-metric_gauge.R (hunk 3 above)
Comment (jthompson-arcus) [reply tests/testthat/test-metric_gauge.R:58]: This is fine as well I suppose.

**Review by jthompson-arcus — APPROVED** [review]
LGTM

**Pull request merged**